set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -560,29 +521,22 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
 	 * @return response entity that comprises response body, headers and status code
 	 * @throws HttpException
 	 */
-	protected ResponseEntity<String> deleteItemFromUserSet(HttpServletRequest request, String wsKey, 
-			String identifier, String datasetId, String localId, String userToken, 
+	protected ResponseEntity<String> deleteItemFromUserSet(HttpServletRequest request, 
+			String identifier, String datasetId, String localId, 
 			String profileStr, String action) throws HttpException {
 		
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
 			//  or if unauthorized respond with HTTP 403
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+		    verifyWriteAccess(Operations.DELETE, request);
```

Comment:
the invokation of the write access needs to be invoked in the calling method

Reply:
verifyWriteAccess invoked as suggested